audio_service/android/src/main/java/com/ryanheise/audioservice/AudioService.java

```diff
@@ -162,22 +163,40 @@ static MediaMetadataCompat getMediaMetadata(String mediaId) {
         return mediaMetadataCache.get(mediaId);
     }
 
-    Bitmap loadArtBitmapFromFile(String path) {
-        Bitmap bitmap = artBitmapCache.get(path);
+    Bitmap loadArtBitmap(String artUriString) {
+        Bitmap bitmap = artBitmapCache.get(artUriString);
         if (bitmap != null) return bitmap;
         try {
+            Uri artUri = Uri.parse(artUriString);
+            boolean usesContentScheme = "content".equals(artUri.getScheme());
+            FileDescriptor fileDescriptor = null;
+            if (usesContentScheme) {
+                fileDescriptor = getContentResolver().openFileDescriptor(artUri, "r").getFileDescriptor();
+            }
```

Comment:
Not going to object to embedding it if it's such a small amount of code, although what's happening with your plugin?

Reply:
There two parts to supporting content URIs - resolving and providing them.

I should mention here that my plugin doesn't support resolving files yet, because that would involve me writing a wrapper for Andoid's FileDescriptor, which is an extra effort I wasn't ready to dedicate time for. But even if it supported that, it would be worse for us to use it here, because instead of just writing a few lines of code from this PR, we would create additional native -> dart -> native hopping.

The other part with providing URIs is in my other local branch, and is blocked by this PR, because enabling ContentProvider means all art URIs will use `content://` scheme. It also will not be required to use the ContentProvider at all - just if one wants other apps to enable resolving URIs from the queue.

Reply:
I forgot to add that the second part is where my plugin will be used

Reply:
> There are two parts to supporting content URIs - resolving and providing them.
> 
> I should mention here that my plugin doesn't support resolving files yet, because that would involve me writing a wrapper for Android's FileDescriptor, which is an extra effort I wasn't ready to dedicate time for. But even if it supported that, it would be worse for us to use it here, because instead of just writing a few lines of code from this PR, we would create additional native -> dart -> native hopping.

Looks like this PR solves the most common use case, which would be to resolve references to the existing media provider.

> The other part with providing URIs is in my other local branch, and is blocked by this PR, because enabling ContentProvider means all art URIs will use `content://` scheme. It also will not be required to use the ContentProvider at all - just if one wants to control how other apps are resolving URIs from the queue.

Got it. So "this" PR can be merged without that one, and that one can be added later. If something is blocked or not working until that other PR is merged, it will help to also link a test / repro case or provide instructions over there on how I can reproduce the issue.

Reply:
> Looks like this PR solves the most common use case, which would be to resolve references to the existing media provider.

Although another plugin is still needed to query the device and return those `content://` URIs. I'm just thinking it would be a good idea to have an example demonstrating this. Was looking at flutter_music_query and judging by the code, it seems to read the image data for you and return that rather than the `content://` URI. Not sure what is the best plugin to demonstrate this. I seem to recall that a file browser plugin allowed you to pick from the music collection on device, but I thought there was also something like flutter_music_query that returned URIs for images.

Reply:
I do have an example for song URIs that I can clean up and post in a separate PR

Reply:
flutter_music_query is not returning art URIs directly, but we can build them ourselves using the song IDs it provides to us

Reply:
Can you include it in this PR? It will be easier to test a new feature when the demo of that feature is in the same PR.

Reply:
I just realized the `flutter_audio_query` is not null-safe - this means that either:
* people would need to use a --no-sound-null-safety flag that example
* we disable null safety for this example and write an example with `// @dart=2.7`

Alernatively, we can use my library, as shown in this example https://github.com/nt4f04uNd/android_content_provider/blob/master/example/lib/main.dart